lib/git_reflow/git_server/pull_request.rb

```diff
@@ -35,13 +53,48 @@ def reviewers_pending_response
         reviewers - approvals
       end
 
+      def check_approvals
+        has_comments_or_approvals = (has_comments? or approvals.any?)
+
+        case self.class.minimum_approvals
+        when "0"
+          true
+        when "", nil
+          # Approvals from every commenter
+          has_comments_or_approvals && reviewers_pending_response.empty?
+        else
+          approvals.size >= self.class.minimum_approvals.to_i and !last_comment.match(self.class.approval_regex).nil?
+        end
+      end
+
+      # Standard Rejection Messages for Error with Reflow Deliver
+      def rejection_message
+        if !build_status.nil? and build_status != "success"
+          "#{build.description}: #{build.target_url}"
+        elsif !approval_minimums_reached?
+          "You need approval from at least #{self.class.minimum_approvals} users!"
+        elsif !all_comments_addressed?
+          # Maybe add what the last comment is?
+          "The last comment is holding up approval:\n#{last_comment}"
+        elsif reviewers_pending_response.count > 0
+          "You still need a LGTM from: #{reviewers_pending_response.join(', ')}"
+        else
+          "Your code has not been reviewed yet."
+        end
+      end
+
+      def approval_minimums_reached?
+        self.class.minimum_approvals.length <= 0 or approvals.size >= self.class.minimum_approvals.to_i
+      end
+
+      def all_comments_addressed?
```

Comment:
@codenamev 
Just a quick note: some of the existing tests are quite flakey and don't always pass all the time. (Tests that I didn't modify, but I noticed that the results would change without modifying any code). Is this of any concern?